x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -869,6 +875,9 @@ class AgentPolicyService {
       await pMap(
         agentPolicies,
         async (agentPolicy) => {
+          const soClient = appContextService.getInternalUserSOClientForSpaceId(
```

Comment:
QQ: since this client is used in many places across the code, could we extract it and export it? It would be easier for other engineers to know that it exists.

I'd also would like to have guidance on when this client is needed, it's not replace all the instance of `soClient` but I'm not sure in which cases I'd need to use this over the base client.

Reply:
> since this client is used in many places across the code, could we extract it and export i

not sure to fully understand It's already exported from the appContextService.

> I'd also would like to have guidance on when this client is needed, it's not replace all the instance of soClient but I'm not sure in which cases I'd need to use this over the base client.

 👍 I will write some dev_docs around all the space awareness caveats in a following PR